.github/workflows/main.yml

```diff
@@ -0,0 +1,32 @@
+name: Tests using qibolab main
+on: push
+jobs:
+  tests:
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install and configure poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      - name: Disable modern installer
+        run: poetry config installer.modern-installation false
+      - name: Install qibolab
+        run: |
+          git clone https://github.com/qiboteam/qibolab
+          cd qibolab
+          poetry install --no-interaction --no-root --with tests --all-extras
+          cd ..
```

Comment:
This is not required at the end of a block (the new shell will re-enter in the default directory).
```suggestion
```